contest/views.py

```diff
@@ -76,7 +85,7 @@ def archive(request, page = None):
     pager = {'previous':previous, 'this':this, 'next':next, 'max_page':max_page}
     return render(request, 
         'contest/contestArchive.html',
-        {'contests':contests,'user':user,'pager':pager},
+        {'contests':contests,'user':user,'groups':groups, 'own_group':own_group,'pager':pager},
```

Comment:
I think it's psoible to use ```if not groups``` to determine whether the list is empty or not. So you might not have to pass ```own_group``` to template file.